api/v1alpha1/eventhub_types.go

```diff
@@ -37,7 +40,9 @@ type EventhubAuthorizationRule struct {
 type EventHubStorageAccount struct {
 	// ResourceGroup - Name of the storage account resource group
 	// +kubebuilder:validation:Pattern=^[-\w\._\(\)]+$
-	ResourceGroup string `json:"resourceGroup,omitempty"`
+	// +kubebuilder:validation:MinLength:1
+	// +kubebuilder:validation:Required
+	ResourceGroup string `json:"resourceGroup"`
```

Comment:
resource group should be optional here. This is becasue the CaptureDescription in the EventthubProperties is optional and since EventHubStorageAccount is a field in CaptureDescription, tthis can be ommitted. This is what is causing the eventhub test in the pipeline to fail.

Reply:
Wow thanks for looking so in depth. Will fix!

Reply:
Fixed this and the other failing tests -- I think it should all pass now.

Reply:
Should we make the StorageAccount field in EventHub Destination a pointer instead so that it can be nil?

Presumably, if the destination is specified, RG is required then right?

Reply:
I've left this alone (not required) for now...

I've filed an issue #1182 for this across the repo though, as I think there are some improvements which could be made.